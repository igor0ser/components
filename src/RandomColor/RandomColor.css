.RandomColor {
  display: flex;
  flex-direction: column;
  width: 300px;
  border: 1px dashed red;
}

.RandomColor__canvas {
  width: 300px;
  height: 200px;
  margin-bottom: 15px;
}