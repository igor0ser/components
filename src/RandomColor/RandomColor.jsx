import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import random from 'lodash/random';
import './RandomColor.css';

class RandomColor extends PureComponent {
  state = { backgroundColor: 'purple' }

  changeColor = () => {
    this.setState({
      backgroundColor:
        `rgb(${random(255)},${random(255)}, ${random(255)})`,
    });
  }

  render() {
    const { header } = this.props;
    const { backgroundColor } = this.state;
    return (
      <div className="RandomColor">
        <h2>{header}</h2>
        <div
          className="RandomColor__canvas"
          style={{ backgroundColor }}
        />
        <button onClick={this.changeColor}>
          Change color
        </button>
      </div>
    );
  }
}

RandomColor.propTypes = {
  header: PropTypes.string.isRequired,
};

export default RandomColor;
