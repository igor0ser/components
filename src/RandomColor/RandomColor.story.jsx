/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import RandomColor from './RandomColor';

const stories = storiesOf('RandomColor', module);
stories.addDecorator(withKnobs);
stories.add('test story', () => {
  const header = text('Text', 'It is test component story');
  return (
    <RandomColor header={header} />
  );
});

