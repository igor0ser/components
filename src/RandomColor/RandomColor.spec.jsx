import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import RandomColor from './RandomColor';

configure({ adapter: new Adapter() });

describe('RandomColor', () => {
  it('should render div.canvas and header', () => {
    const comp = shallow(<RandomColor header="test" />);

    expect(comp.find('h2').text()).toBe('test');
    expect(toJson(comp)).toMatchSnapshot();
  });
});
