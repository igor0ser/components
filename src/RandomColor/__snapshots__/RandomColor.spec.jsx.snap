// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor should render div.canvas and header 1`] = `
<div
  className="RandomColor"
>
  <h2>
    test
  </h2>
  <div
    className="RandomColor__canvas"
    style={
      Object {
        "backgroundColor": "purple",
      }
    }
  />
  <button
    onClick={[Function]}
  >
    Change color
  </button>
</div>
`;
